OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1645,16 +1670,9 @@ int Model::replaceMarkerSet(const SimTK::State& s, MarkerSet& aMarkerSet)
     // Now add the markers from aMarkerSet whose body names match bodies in the engine.
     for (i = 0; i < aMarkerSet.getSize(); i++)
     {
-        // Eran: we make a *copy* since both _markerSet and aMarkerSet own their elements (so they will delete them)
-        Marker* marker = aMarkerSet.get(i).clone();
-        const string& frameName = marker->getFrameName();
-        if (getFrameSet().contains(frameName))
-        {
-            const OpenSim::PhysicalFrame* frame = dynamic_cast<const PhysicalFrame*>(&getFrameSet().get(frameName));
-            if(frame) marker->changeFrame(*frame);
-            upd_MarkerSet().adoptAndAppend(marker);
-            numAdded++;
-        }
+        Marker* marker = aMarkerSet[i].clone();
+        upd_MarkerSet().adoptAndAppend(marker);
+        ++numAdded;
     }
```

Comment:
Yes. We don't rely on the `Model` using this method to both replace the MarkererSet (editing the model) and forming the connections. We rely on `finalizeConnections()` to validate and throw if parent frames cannot be found. I think that is more robust than simply dropping Markers and more flexible, since users could still be adding Frames to the Model, in which case it is premature to drop markers.